libraries/protocol/custom_authority.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2018 Abit More, and contributors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <graphene/protocol/custom_authority.hpp>
+#include <graphene/protocol/operations.hpp>
+#include <graphene/protocol/restriction_predicate.hpp>
+
+#include <fc/io/raw.hpp>
+
+namespace graphene { namespace protocol {
+
+share_type custom_authority_create_operation::calculate_fee(const fee_parameters_type& k)const {
+   share_type core_fee_required = k.basic_fee;
+   core_fee_required += k.price_per_byte * (fc::raw::pack_size(restrictions) + fc::raw::pack_size(auth));
```

Comment:
Note: `price_per_byte` is 32 bit; `fc::raw::pack_size()` returns `size_t` which is 64 bit, but practically the result won't be greater than 32 bit; so the result of the operator `*` will be 64 bit and won't overflow. The operator `+=` of `share_type` is safe too.